### PR TITLE
JAMES-2470 Various backend reuse improvments

### PR DESCRIPTION
**Pull request by chibenwa**
Some small improvements:

 - Allow HBase cluster reuse across classes
 - Allow Tika docker container reuse across class

```
[INFO] Apache James :: Mailbox :: ElasticSearch ........... SUCCESS [01:28 min]
[INFO] Apache James :: Mailbox :: HBase ................... SUCCESS [01:22 min]
[INFO] Apache James :: Server :: Data :: HBase Persistence  SUCCESS [01:21 min]
```

After:

```
[INFO] Apache James :: Mailbox :: ElasticSearch ........... SUCCESS [01:02 min]
[INFO] Apache James :: Mailbox :: HBase ................... SUCCESS [ 17.385 s]
[INFO] Apache James :: Server :: Data :: HBase Persistence  SUCCESS [ 27.267 s]
```

Gain: **2 min 23**

### CODE REVIEW

File: mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessageTest.java [hunk 1]
```diff
@@ -61,7 +61,7 @@
     public static final MessageUid MESSAGE_UID = MessageUid.of(154);
 
     @ClassRule
-    public static TikaContainer tika = new TikaContainer();
+    public static TikaContainerStaticRule tika = TikaContainerStaticRule.rule;
```
Comment (trantienduchn) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessageTest.java]: how do you stop docker tika after tests?

Comment (chibenwa) [reply mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessageTest.java]: It's testcontainers resource reaper which ensures that.

But you are right. I need at least a comment.

File: mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TikaContainerStaticRule.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.tika;
+
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+public class TikaContainerStaticRule implements TestRule {
```
Comment (mbaechler) [inline mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TikaContainerStaticRule.java]: What does Static mean here ?

File: mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TikaContainerStaticRule.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    public static final TikaContainerStaticRule rule = new TikaContainerStaticRule(new TikaContainer());
+
+    private final TikaContainer tikaContainer;
+
+    public TikaContainerStaticRule(TikaContainer tikaContainer) {
```
Comment (mbaechler) [inline mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TikaContainerStaticRule.java]: private ?

File: mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TikaContainerStaticRule.java (hunk 2 above)
Comment (chibenwa) [reply mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TikaContainerStaticRule.java]: Singleton I guess

File: mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessageTest.java [hunk 4]
```diff
@@ -61,7 +61,7 @@
     public static final MessageUid MESSAGE_UID = MessageUid.of(154);
 
     @ClassRule
```
Comment (bsideup) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessageTest.java]: actually, you don't need a ClassRule here at all :)
```
public static TikaContainer tika = new TikaContainer();
static {
   tika.start();
}
```

will give you the same result :)

Comment (chibenwa) [reply mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessageTest.java]: Hi!

First thanks for the feedback.

We did discuss that in detail out here https://github.com/linagora/james-project/pull/1584#discussion_r205094601 . The outcome is that we want to hide the fact that we are using a singleton under the hood, and ease future migration back to JUNIT4 rules.

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**